src/modules/claims/claims.service.ts

```diff
@@ -392,8 +398,16 @@ export class ClaimsService {
     const strippedClaimData = this.stripClaimData(claimData);
     const { claimType: role, claimTypeVersion: version } = claimData;
 
+    const defaultValidityPeriod = roleDefinition?.['defaultValidityPeriod'];
+    const roleDefinitionExpirationTimestamp = defaultValidityPeriod
+      ? Date.now() + defaultValidityPeriod
+      : undefined;
+
+    const claimExpirationTimestamp =
+      expirationTimestamp || roleDefinitionExpirationTimestamp;
+
     if (registrationTypes.includes(RegistrationTypes.OnChain)) {
-      const expiry = expirationTimestamp || defaultClaimExpiry;
+      const expiry = claimExpirationTimestamp || defaultClaimExpiry;
```

Comment:
Not related to this PR, but, I wonder is it worth replace `defaultClaimExpiry` by `Erternity` ?

Reply:
It just a new name. To express expiry which never expires

Reply:
Ohh okay, make sense.

Reply:
@JGiter @Harasz We can remove the `defaultClaimExpiry`. IMO either the expiry should come from the `RoleDefinition` or the `issuer` itself. If let's say the issuer wants it to be non-expiry claim, then they themselves should pass this a value.

Reply:
@jrhender what do you think about it?

Reply:
I'm open to the rename to `eternity`

Regarding removing the `defaultClaimExpiry`, I think we need it because the contract requires expiry https://github.com/energywebfoundation/ew-credentials/blob/c54672118a2687146e8421d3779208a04ee191c0/packages/onchain-claims/contracts/ClaimManager.sol#L93